search/searcher/search_geoshape.go

```diff
@@ -70,6 +70,20 @@ func buildRelationFilterOnShapes(ctx context.Context, dvReader index.DocValueRea
 	var dvShapeValue []byte
 	var startReading, finishReading bool
 	var reader *bytes.Reader
+
+	bufPool := make([][]byte, 11)
+	bufPool[0] = make([]byte, 8192*3)
```

Comment:
since the buffer sizes follow a step like fashion, maybe defining a constant with respect to the biggest buffer size and allocating a buf, buf/2, buf/4 etc. amount of space looks more readable